cpp/src/io/utilities/column_type_histogram.hpp

```diff
@@ -33,6 +33,11 @@ struct column_type_histogram {
   cudf::size_type positive_small_int_count{};
   cudf::size_type big_int_count{};
   cudf::size_type bool_count{};
+  auto total_count() const
```

Comment:
Perhaps for the future -- seems like this struct would be more versatile and more easily extendible (and totals easier to calculate in a bug-free way) if you store a `std::array` (or similar) of counts and index it using constant names for indices. Then this line could just be a `std::accumulate` (or similar).